Content.Server/GameTicking/Rules/ZombieRuleSystem.cs

```diff
@@ -180,12 +293,58 @@ private void OnStartAttempt(RoundStartAttemptEvent ev)
     protected override void Started(EntityUid uid, ZombieRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
     {
         base.Started(uid, component, gameRule, args);
-        InfectInitialPlayers(component);
+
+        component.AnnounceAt = GameTicker.RoundDuration() +
+                               TimeSpan.FromSeconds(_random.NextFloat(component.AnnounceMin,
+                                   component.AnnounceMax));
+        component.InfectInitialAt = GameTicker.RoundDuration() + TimeSpan.FromSeconds(component.InitialInfectDelaySecs);
+
+        component.FirstTurnAllowed = GameTicker.RoundDuration() + TimeSpan.FromSeconds(component.TurnTimeMin);
+
+        if (component.EarlySettings.EmoteSoundsId != null)
+        {
+            _prototypeManager.TryIndex(component.EarlySettings.EmoteSoundsId, out component.EarlySettings.EmoteSounds);
+        }
+        if (component.VictimSettings.EmoteSoundsId != null)
+        {
+            _prototypeManager.TryIndex(component.VictimSettings.EmoteSoundsId, out component.VictimSettings.EmoteSounds);
+        }
     }
 
-    private void OnZombifySelf(EntityUid uid, ZombifyOnDeathComponent component, ZombifySelfActionEvent args)
+    protected override void ActiveTick(EntityUid uid, ZombieRuleComponent component, GameRuleComponent gameRule,
+        float frameTime)
     {
-        _zombify.ZombifyEntity(uid);
+        if (component.InfectInitialAt != TimeSpan.Zero && component.InfectInitialAt < GameTicker.RoundDuration())
+        {
+            // Time to infect the initial players
+            InfectInitialPlayers(uid, component);
+            component.InfectInitialAt = TimeSpan.Zero;
+        }
+
+        if (component.AnnounceAt != TimeSpan.Zero && component.AnnounceAt < GameTicker.RoundDuration())
+        {
+            // Announce the brain eating about to commence
+            if (_random.Prob(component.AnnounceChance))
+            {
+                _chatManager.DispatchServerAnnouncement(Loc.GetString("zombies-will-eat-your-brains"));
+            }
+            component.AnnounceAt = TimeSpan.Zero;
+        }
+
+        if (component.FirstTurnAllowed != TimeSpan.Zero && component.FirstTurnAllowed < GameTicker.RoundDuration())
+        {
+            ActivateZombifyOnDeath(uid, component);
+            component.FirstTurnAllowed = TimeSpan.Zero;
+        }
+    }
```

Comment:
Just checking roundduration will break with deserialization.

Reply:
Could you explain how and propose an alternative? Does the time essentially jump?

Reply:
I have changed this to current time, I hope that's sufficient?